Pytest ini hijacking normal Pytest functioning

Once again, I am coming back to working on PrOMMiS after working on other IDAES ecosystem projects, and I am once again frustrated that Pytest doesn't work like it does in every other project. The way Pytest works by default is, if you call `pytest tests\test_uky_flowsheet.py`, it will just collect and run the tests in `test_uky_flowsheet.py`. However, if you do this in PrOMMiS, it collects and runs *every test in the repository*. The workaround suggested in the README is to call `pytest -k tests\test_uky_flowsheet.py`. ~~However, it *still* collects every test in the repository but skips every test that's not in `test_uky_flowsheet.py`. If there is an error in a file in a totally different directory, it causes Pytest to raise an exception and terminate:~~

```
(prommis) [REDACTED]\Repos\prommis\src\prommis\uky>pytest -k tests\test_uky_flowsheet.py
================================================= test session starts =================================================
platform win32 -- Python 3.11.13, pytest-7.4.4, pluggy-1.6.0
rootdir: [REDACTED]\Repos\prommis
configfile: pyproject.toml
plugins: anyio-4.10.0, idaes-flowsheet-processor-0.2, nbmake-1.4.6, prommis-0.9.0+g532de23c8, cov-7.0.0
collected 605 items / 2 errors / 605 deselected / 0 selected

======================================================= ERRORS ========================================================
_____________________ ERROR collecting src/prommis/costing/uq/tests/test_diafiltration_cost_uq.py _____________________
ImportError while importing test module '[REDACTED]\Repos\prommis\src\prommis\costing\uq\tests\test_diafiltration_cost_uq.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
..\costing\uq\tests\test_diafiltration_cost_uq.py:36: in <module>
    import prommis.costing.uq.diafiltration_cost_uq as uq
..\costing\uq\diafiltration_cost_uq.py:36: in <module>
    from idaes.core.util.model_diagnostics import DiagnosticsToolbox, degrees_of_freedom
E   ImportError: cannot import name 'degrees_of_freedom' from 'idaes.core.util.model_diagnostics' ([REDACTED]\Repos\idaes-pse\idaes\core\util\model_diagnostics.py)
_________ ERROR collecting src/prommis/uky/costing/tests/test_diafiltration_flowsheet_optimization_example.py _________
ImportError while importing test module '[REDACTED]\Repos\prommis\src\prommis\uky\costing\tests\test_diafiltration_flowsheet_optimization_example.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
costing\tests\test_diafiltration_flowsheet_optimization_example.py:27: in <module>
    from idaes.core.util.model_diagnostics import DiagnosticsToolbox, degrees_of_freedom
E   ImportError: cannot import name 'degrees_of_freedom' from 'idaes.core.util.model_diagnostics' ([REDACTED]\Repos\idaes-pse\idaes\core\util\model_diagnostics.py)
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 2 errors during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
========================================== 605 deselected, 2 errors in 1.99s ==========================================
```
Edit: using the `-k` option now seems to fail, as mentioned in https://github.com/prommis/prommis/pull/178#discussion_r2350239365.

(As a side note, why is this error not getting picked up in our nightly CI?)

Additionally, if you want to run every test in a directory, the default method is to run Pytest on that directory: `pytest tests`. However, that does not work and the `-k` option does not allow it to work. #178 has a workaround that involves overwriting the `.ini`, but why should we have to fight the `.ini` file every time we want to carry out routine operations using Pytest. Furthermore, it *also* breaks calling Pytest from an IDE, because the IDE also expects Pytest to behave normally.

Why does PrOMMiS need an `.ini` file different than either IDAES or WaterTAP? Is there any other way to accomplish what the current `.ini` is accomplishing?

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -101,7 +101,6 @@ known_issues = "prommis._testing_utils.known_issues"
 
 [tool.pytest.ini_options]
 addopts = """
---pyargs prommis
 --import-mode=importlib
 --cov=prommis
 --cov-config .coveragerc

